libs/blocks/modal/modal.js

```diff
@@ -202,15 +199,17 @@ export async function getModal(details, custom) {
       .forEach((element) => element.setAttribute('aria-disabled', 'true'));
   }
   if (dialog.classList.contains('commerce-frame')) {
-    if (isInitialPageLoad) {
-      window.addEventListener('message', (messageInfo) => {
-        if (dialog.classList.contains('height-fit-content')) {
-          adjustModalHeight({ contentHeight: messageInfo?.data?.contentHeight, dialogId: id });
-        }
-        sendViewportDimensionsOnRequest(messageInfo);
-      });
-      isInitialPageLoad = false;
-    }
+    const { debounce } = await import('../../utils/action.js');
+    messageAbortController = new AbortController();
+    window.addEventListener('message', (messageInfo) => {
+      if (dialog.classList.contains('height-fit-content')) {
+        adjustModalHeight({ contentHeight: messageInfo?.data?.contentHeight, dialog });
+      }
+      /* If the page inside iFrame comes from another domain, it won't be able to retrieve
+      the viewport dimensions, so it sends a request to receive the viewport dimensions
+      from the parent window. */
+      sendViewportDimensionsOnRequest({ debounce, messageInfo });
```

Comment:
If the user opens first modal, then closes it and opens another one, won't this result in multiple active `resize` listeners in the parent window?